### PR TITLE
Move download button for media to the action bar

### PR DESCRIPTION
**Pull request by turt2live**
Fixes: https://github.com/vector-im/element-web/issues/17943

----

**Code reviewer**: This is broken out into chunked commits, though somewhat poorly (sorry).

**Dev note**: The differences in style are known and out of scope for this PR. See https://github.com/vector-im/element-web/issues/18025 for details.

----

Showing the download button and lack of links below media events:
image

Download/decrypt in progress:
image

Image example:
image

Video files weren't forgotten, but not shown here. It's an image with a browser-default player.

<details>
<summary>To show that it still works fine in the right panel, here's a Files Panel</summary>

image

</details>

### CODE REVIEW

**Comment by turt2live** [comment]
There were problems with replies during development, so just highlighting that it seems fine now:
image

**Review by gaelledel — COMMENTED** [review]
- [ ] New download button looks good

Screenshot 2021-07-19 at 12 01 47
Screenshot 2021-07-19 at 12 02 06

- [ ] Can we remove the Hover state (white background on Light theme and Black bg on Dark theme) it makes the UI look broken.
- [ ] On the Desktop version, the Component style hasn't been updated to the new styling. 
Screenshot 2021-07-19 at 12 01 47

**Comment by turt2live** [comment]
> Can we remove the Hover state (white background on Light theme and Black bg on Dark theme) it makes the UI look broken.

Will do so in another PR.

> On the Desktop version, the Component style hasn't been updated to the new styling.

It hasn't quite been released yet, but soon! Will be looking at exposing it properly over the next few days.

**Review by gaelledel — APPROVED** [review]

File: src/components/views/messages/DownloadActionButton.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        const blob = await this.props.mediaEventHelperGet().sourceBlob.value;
+        this.setState({ blob });
+    };
+
+    private onFrameLoad = () => {
+        this.setState({ loading: false });
+
+        // we aren't showing the iframe, so we can send over the bare minimum styles and such.
+        this.iframe.current.contentWindow.postMessage({
+            imgSrc: DOWNLOAD_ICON_URL,
```
Comment (dbkr) [inline src/components/views/messages/DownloadActionButton.tsx:72]: Looks like you're passing an image to the iframe which seems odd if it's always display: none?

File: src/utils/MediaEventHelper.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import { EventType, MsgType } from "matrix-js-sdk/src/@types/event";
+
+// TODO: We should consider caching the blobs. https://github.com/vector-im/element-web/issues/17192
+
+export class MediaEventHelper implements IDestroyable {
+    public readonly sourceUrl: LazyValue<string>;
+    public readonly thumbnailUrl: LazyValue<string>;
+    public readonly sourceBlob: LazyValue<Blob>;
+    public readonly thumbnailBlob: LazyValue<Blob>;
+    public readonly media: Media;
```
Comment (dbkr) [inline src/utils/MediaEventHelper.ts:32]: These are pretty obvious now I've read the code but some comments wouldn't go amiss (my confusion was mostly whether the source blob/url was decrypted or undecrypted).

**Review by dbkr — APPROVED** [review]
looks plausible I think

**Review by dbkr — APPROVED** [review]
thanks!

**Pull request merged**